backend/owa.ts

```diff
@@ -25,31 +32,52 @@ export async function fetchSessionData(partition: string, url: string, interacti
           return;
         }
         finished = true;
-        session.cookies.removeListener('changed', checkCanary);
+        session.cookies.removeListener('changed', onCookie);
+        if (isHotmail) {
+          data.owaURL = url;
+        }
```

Comment:
`data` will be null if the user closes the window instead of logging in. I also found it confusing that you're using the value of `url` set below in `checkLoginFinished`. Could this be moved into `checkLoginFinished`?

Reply:
Ok, thanks for the catch.